### PR TITLE
fix(@types/node): markdown formatting mistakes

### PR DESCRIPTION
**Pull request by forivall**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- N/A ~~Provide a URL to documentation or source code which provides context for the suggested changes:~~
- N/A ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

---

Logically, this is an extension of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59930

### CODE REVIEW

File: types/node/scripts/generate-docs/html-doc-processing.ts [hunk 1]
```diff
@@ -120,6 +120,9 @@ export function fixupHtmlDocs(input: string, context: DocContext): string {
         input = input.replace(/<pre><code class="language-cjs">.*?<\/code><\/pre>/igms, '');
     }
 
+    // workaround markdown generation bug
+    input = input.replaceAll(/([^ ])\n<em>/g, '$1 \n<em>');
```
Comment (gabritto) [inline types/node/scripts/generate-docs/html-doc-processing.ts:124]: What's this script used for and what is this line supposed to do?

Comment (peterblazejewicz) [reply types/node/scripts/generate-docs/html-doc-processing.ts:124]: @SimonSchick Simon is the original owner of the implementation (sorry for direct ping Simon)

Comment (SimonSchick) [reply types/node/scripts/generate-docs/html-doc-processing.ts:124]: I think this was just suppose to strip whitespaces, I can't remember the exact logic here..

**Review by andrewbranch — APPROVED** [review]
Thanks!

File: types/node/scripts/generate-docs/html-doc-processing.ts (hunk 1 above)
Comment (forivall) [reply types/node/scripts/generate-docs/html-doc-processing.ts:124]: This line detects where there's a `<newline>` `<em>` that isnt preceded by a space, and adds a space. The markdown generation seems to drop the spacing when it's a newline, so this is a workaround.

**Comment by forivall** [comment]
Ready to Merge

**Pull request merged**